homeassistant/components/blue_current/__init__.py

```diff
@@ -0,0 +1,193 @@
+"""The Blue Current integration."""
+from __future__ import annotations
+
+from contextlib import suppress
+from datetime import datetime
+from typing import Any
+
+from bluecurrent_api import Client
+from bluecurrent_api.exceptions import (
+    BlueCurrentException,
+    InvalidApiToken,
+    RequestLimitReached,
+    WebsocketError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_NAME,
+    CONF_API_TOKEN,
+    EVENT_HOMEASSISTANT_STOP,
+    Platform,
+)
+from homeassistant.core import Event, HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_call_later
+
+from .const import DOMAIN, EVSE_ID, LOGGER, MODEL_TYPE
+
+PLATFORMS = [Platform.SENSOR]
+CHARGE_POINTS = "CHARGE_POINTS"
+DATA = "data"
+SMALL_DELAY = 1
+LARGE_DELAY = 20
+
+GRID = "GRID"
+OBJECT = "object"
+VALUE_TYPES = ["CH_STATUS"]
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up Blue Current as a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    client = Client()
+    api_token = config_entry.data[CONF_API_TOKEN]
+    connector = Connector(hass, config_entry, client)
+    try:
+        await connector.connect(api_token)
+    except InvalidApiToken:
+        LOGGER.error("Invalid Api token")
+        return False
+    except BlueCurrentException as err:
+        raise ConfigEntryNotReady from err
+
+    hass.loop.create_task(connector.start_loop())
+    await client.get_charge_points()
+
+    await client.wait_for_response()
+    hass.data[DOMAIN][config_entry.entry_id] = connector
+    await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)
+
+    async def _async_disconnect_websocket(_: Event) -> None:
+        await connector.disconnect()
+
+    config_entry.async_on_unload(
+        hass.bus.async_listen_once(
+            EVENT_HOMEASSISTANT_STOP, _async_disconnect_websocket
+        )
+    )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Unload the Blue Current config entry."""
+    connector: Connector = hass.data[DOMAIN].pop(config_entry.entry_id)
+    hass.async_create_task(connector.disconnect())
```

Comment:
I think you can remove this and just disconnect the websocket on unload `entry.async_on_unload(_async_disconnect_websocket)`. This way you don't have to call this here